service_capacity_modeling/interface.py

```diff
@@ -333,12 +335,43 @@ def family(self):
     def size(self):
         return self.name.rsplit(self.family_separator, 1)[1]
 
+    @property
+    def normalized_size(self) -> Fraction:
+        numeric = re.findall(r"\d+", self.size)
+        if numeric:
+            assert len(numeric) == 1
+            return Fraction(float(numeric[0]))
+        return {
+            "medium": Fraction(1, 4),
+            "large": Fraction(1, 2),
+            "xlarge": Fraction(1),
+        }.get(self.size)
+
     @staticmethod
     def get_managed_instance() -> Instance:
         return Instance(
             name="managed.0", cpu=0, cpu_ghz=0, ram_gib=0, net_mbps=0, drive=None
         )
 
+    def merge_with(self, overrides: "Instance") -> "Instance":
+        self_dict = self.model_dump()
+        other_dict = overrides.model_dump(exclude_unset=True)
+
+        for (k, v) in other_dict.items():
+            # TODO we need a deep merge on drive (recursive merge)
+            if k in ("platforms",):
+                # Unique merge platforms
+                merged_platforms = list(self_dict.get("platforms", []))
```

Comment:
how about

```
merged_platforms = list(
  set(self_dist.get("platforms", [])) | set(other_dict.get("platforms", []))
)
```

Reply:
I'm going to pull in the nflx-configs deep merge functionality when I add layering, which can deep merge arbitrary structured objects with override support.